src/pager.rs

```diff
@@ -0,0 +1,120 @@
+use shell_words::ParseError;
+use std::env;
+
+/// If we use a pager, this enum tells us from where we were told to use it.
+#[derive(Debug, PartialEq)]
+pub enum PagerSource {
+    /// From --config
+    Config,
+
+    /// From the env var BAT_PAGER
+    BatPagerEnvVar,
+
+    /// From the env var PAGER
+    PagerEnvVar,
+
+    /// No pager was specified, default is used
+    Default,
+}
+
+/// We know about some pagers, for example 'less'. This is a list of all pagers we know about
+#[derive(Debug, PartialEq)]
+pub enum PagerKind {
+    /// The pager is ourselves
+    Bat,
+
+    /// less
+    Less,
+
+    /// more
+    More,
+
+    /// most
+    Most,
+
+    /// A pager we don't know about
+    Unknown,
+}
+
+impl PagerKind {
+    fn from_bin(bin: &str) -> PagerKind {
+        use std::ffi::OsStr;
+        use std::path::Path;
+
+        let stem = Path::new(bin)
+            .file_stem()
+            .unwrap_or_else(|| OsStr::new("unknown"));
+
+        if stem == OsStr::new("bat") {
+            PagerKind::Bat
+        } else if stem == OsStr::new("less") {
+            PagerKind::Less
+        } else if stem == OsStr::new("more") {
+            PagerKind::More
+        } else if stem == OsStr::new("most") {
+            PagerKind::Most
+        } else {
+            PagerKind::Unknown
+        }
+    }
+}
+
+/// A pager such as 'less', and from where we got it.
+#[derive(Debug)]
+pub struct Pager {
+    // The pager binary
+    pub bin: String,
+
+    // The pager binary arguments (that we might tweak)
+    pub args: Vec<String>,
+
+    // What pager this is
+    pub kind: PagerKind,
+
+    // From where this pager comes
```

Comment:
I'll replace them by real doc-strings, it's just a typo on my part. I prefer to leave them in since some of them do add some extra value I believe (e.g. `// The pager binary arguments (that we might tweak)`, and the asymmetry of missing docs on some fields is more annoying than "obvious" docs IMHO, but that's mostly a matter of taste I guess. If you change your mind and strongly want to me remove the "obvious" docs, no worries, just let me know :)